### ISSUE TITLE
AddressSanitizer: stack-buffer-overflow in testdrive/sqllogictest in to_standard_notation_string

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#25634]
### What version of Materialize are you using?

main

### What is the issue?

```
=================================================================
==50099==ERROR: AddressSanitizer: stack-buffer-overflow on address 0xffff952644a4 at pc 0xaaaad304b2dc bp 0xfffff4337bb0 sp 0xfffff4337bc8
READ of size 2 at 0xffff952644a4 thread T0
    #0 0xaaaad304b2d8 in decNumberGetBCD (/usr/local/bin/testdrive+0xcd6b2d8) (BuildId: 2bba89f41fc100da)
    #1 0xaaaad0f60f78 in dec::decimal::Decimal$LT$_$GT$::to_standard_notation_string::hf35d5f77ef63d672 (/usr/local/bin/testdrive+0xac80f78) (BuildId: 2bba89f41fc100da)
    #2 0xaaaad0a86c50 in _$LT$mz_testdrive..action..sql..decode_row..NumericStandardNotation$u20$as$u20$core..fmt..Display$GT$::fmt::h0f63f4e9f9113dbe (/usr/local/bin/testdrive+0xa7a6c50) (BuildId: 2bba89f41fc100da)
    #3 0xaaaad0a7785c in mz_testdrive::action::sql::decode_row::h6e928c57661004d9 (/usr/local/bin/testdrive+0xa79785c) (BuildId: 2bba89f41fc100da)
    #4 0xaaaad11b8f98 in _$LT$core..iter..adapters..map..Map$LT$I$C$F$GT$$u20$as$u20$core..iter..traits..iterator..Iterator$GT$::try_fold::h83412cc359ccf0be (/usr/local/bin/testdrive+0xaed8f98) (BuildId: 2bba89f41fc100da)
    #5 0xaaaad0839d60 in _$LT$core..iter..adapters..GenericShunt$LT$I$C$R$GT$$u20$as$u20$core..iter..traits..iterator..Iterator$GT$::try_fold::h40c875663c3c0361 (/usr/local/bin/testdrive+0xa559d60) (BuildId: 2bba89f41fc100da)
    #6 0xaaaad0c03df0 in alloc::vec::in_place_collect::_$LT$impl$u20$alloc..vec..spec_from_iter..SpecFromIter$LT$T$C$I$GT$$u20$for$u20$alloc..vec..Vec$LT$T$GT$$GT$::from_iter::h7a0f6f5dbfa2d094 (/usr/local/bin/testdrive+0xa923df0) (BuildId: 2bba89f41fc100da)
    #7 0xaaaad08aa1c8 in core::iter::adapters::try_process::hf1bff90039b34b06 (/usr/local/bin/testdrive+0xa5ca1c8) (BuildId: 2bba89f41fc100da)
    #8 0xaaaad0b72c18 in _$LT$tokio..time..timeout..Timeout$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h470bea1cf504a486 (/usr/local/bin/testdrive+0xa892c18) (BuildId: 2bba89f41fc100da)
    #9 0xaaaad039e66c in _$LT$mz_testdrive..parser..PosCommand$u20$as$u20$mz_testdrive..action..Run$GT$::run::_$u7b$$u7b$closure$u7d$$u7d$::h69a822f38ea0378c (/usr/local/bin/testdrive+0xa0be66c) (BuildId: 2bba89f41fc100da)
    #10 0xaaaad2160738 in mz_testdrive::run_line_reader::_$u7b$$u7b$closure$u7d$$u7d$::hd47d6711740033b9 (/usr/local/bin/testdrive+0xbe80738) (BuildId: 2bba89f41fc100da)
    #11 0xaaaad214f738 in mz_testdrive::run_string::_$u7b$$u7b$closure$u7d$$u7d$::hac94137da0155857 (/usr/local/bin/testdrive+0xbe6f738) (BuildId: 2bba89f41fc100da)
    #12 0xaaaad21b64b8 in testdrive::main::_$u7b$$u7b$closure$u7d$$u7d$::h91f1ab587e7a2b0e (/usr/local/bin/testdrive+0xbed64b8) (BuildId: 2bba89f41fc100da)
    #13 0xaaaad21a4a5c in tokio::runtime::park::CachedParkThread::block_on::he4c858825cfc38bb (/usr/local/bin/testdrive+0xbec4a5c) (BuildId: 2bba89f41fc100da)
    #14 0xaaaad252b6b8 in tokio::runtime::runtime::Runtime::block_on::hf30ef8dec6f955cb (/usr/local/bin/testdrive+0xc24b6b8) (BuildId: 2bba89f41fc100da)
    #15 0xaaaad23c5ccc in testdrive::main::hd708545af303b2f5 (/usr/local/bin/testdrive+0xc0e5ccc) (BuildId: 2bba89f41fc100da)
    #16 0xaaaad23cd398 in std::sys_common::backtrace::__rust_begin_short_backtrace::h951b52443384962d (/usr/local/bin/testdrive+0xc0ed398) (BuildId: 2bba89f41fc100da)
    #17 0xaaaad25ecfd8 in std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::h39b9042ba2fe0802 (/usr/local/bin/testdrive+0xc30cfd8) (BuildId: 2bba89f41fc100da)
    #18 0xaaaad20f7f5c in std::rt::lang_start_internal::h4521b8f735c65f30 (/usr/local/bin/testdrive+0xbe17f5c) (BuildId: 2bba89f41fc100da)
    #19 0xaaaad25ecf5c in std::rt::lang_start::h17f212e5979d9865 (/usr/local/bin/testdrive+0xc30cf5c) (BuildId: 2bba89f41fc100da)
    #20 0xaaaad23c5ee4 in main (/usr/local/bin/testdrive+0xc0e5ee4) (BuildId: 2bba89f41fc100da)
    #21 0xffff966b73f8  (/lib/aarch64-linux-gnu/libc.so.6+0x273f8) (BuildId: 3ba44e06b9dc66aeeb2651db4dd015ffaf6e0849)
    #22 0xffff966b74c8 in __libc_start_main (/lib/aarch64-linux-gnu/libc.so.6+0x274c8) (BuildId: 3ba44e06b9dc66aeeb2651db4dd015ffaf6e0849)
    #23 0xaaaac954b02c in _start (/usr/local/bin/testdrive+0x326b02c) (BuildId: 2bba89f41fc100da)

Address 0xffff952644a4 is located in stack of thread T0 at offset 17572 in frame
    #0 0xaaaad0a6e50c in mz_testdrive::action::sql::decode_row::h6e928c57661004d9 (/usr/local/bin/testdrive+0xa78e50c) (BuildId: 2bba89f41fc100da)

  This frame has 324 object(s):
    [32, 96) 'formatter.i1964'
    [128, 152) 'buf.i1965'
    [192, 256) 'formatter.i1948'
    [288, 312) 'buf.i1949'
    [352, 416) 'formatter.i1933'
    [448, 472) 'buf.i1934'
    [512, 528) '_40.i.i.i1871'
    [544, 592) '_34.i.i.i1872'
    [624, 628) '_25.i.i.i1873'
    [640, 672) '_20.i.i.i1874'
    [704, 752) '_16.i.i.i1875'
    [784, 848) 'formatter.i.i1876'
    [880, 904) 'buf.i.i1877'
    [944, 1008) 'formatter.i1856'
    [1040, 1064) 'buf.i1857'
    [1104, 1120) '_40.i.i.i1795'
    [1136, 1184) '_34.i.i.i1796'
    [1216, 1220) '_25.i.i.i1797'
    [1232, 1264) '_20.i.i.i1798'
    [1296, 1344) '_16.i.i.i1799'
    [1376, 1440) 'formatter.i.i1800'
    [1472, 1496) 'buf.i.i1801'
    [1536, 1600) 'formatter.i1780'
    [1632, 1656) 'buf.i1781'
    [1696, 1712) '_40.i.i.i1719'
    [1728, 1776) '_34.i.i.i1720'
    [1808, 1812) '_25.i.i.i1721'
    [1824, 1856) '_20.i.i.i1722'
    [1888, 1936) '_16.i.i.i1723'
    [1968, 2032) 'formatter.i.i1724'
    [2064, 2088) 'buf.i.i1725'
    [2128, 2192) 'formatter.i1704'
    [2224, 2248) 'buf.i1705'
    [2288, 2304) '_40.i.i.i1643'
    [2320, 2368) '_34.i.i.i1644'
    [2400, 2404) '_25.i.i.i1645'
    [2416, 2448) '_20.i.i.i1646'
    [2480, 2528) '_16.i.i.i1647'
    [2560, 2624) 'formatter.i.i1648'
    [2656, 2680) 'buf.i.i1649'
    [2720, 2784) 'formatter.i1628'
    [2816, 2840) 'buf.i1629'
    [2880, 2896) '_40.i.i.i1567'
    [2912, 2960) '_34.i.i.i1568'
    [2992, 2996) '_25.i.i.i1569'
    [3008, 3040) '_20.i.i.i1570'
    [3072, 3120) '_16.i.i.i1571'
    [3152, 3216) 'formatter.i.i1572'
    [3248, 3272) 'buf.i.i1573'
    [3312, 3376) 'formatter.i1552'
    [3408, 3432) 'buf.i1553'
    [3472, 3488) '_40.i.i.i1491'
    [3504, 3552) '_34.i.i.i1492'
    [3584, 3588) '_25.i.i.i1493'
    [3600, 3632) '_20.i.i.i1494'
    [3664, 3712) '_16.i.i.i1495'
    [3744, 3808) 'formatter.i.i1496'
    [3840, 3864) 'buf.i.i1497'
    [3904, 3968) 'formatter.i1476'
    [4000, 4024) 'buf.i1477'
    [4064, 4080) '_40.i.i.i1415'
    [4096, 4144) '_34.i.i.i1416'
    [4176, 4180) '_25.i.i.i1417'
    [4192, 4224) '_20.i.i.i1418'
    [4256, 4304) '_16.i.i.i1419'
    [4336, 4400) 'formatter.i.i1420'
    [4432, 4456) 'buf.i.i1421'
    [4496, 4560) 'formatter.i.i1399'
    [4592, 4616) 'buf.i.i1400'
    [4656, 4672) '_40.i.i.i1338'
    [4688, 4736) '_34.i.i.i1339'
    [4768, 4772) '_25.i.i.i1340'
    [4784, 4816) '_20.i.i.i1341'
    [4848, 4896) '_16.i.i.i1342'
    [4928, 4992) 'formatter.i.i1343'
    [5024, 5048) 'buf.i.i1344'
    [5088, 5152) 'formatter.i1323'
    [5184, 5208) 'buf.i1324'
    [5248, 5312) 'formatter.i1308'
    [5344, 5368) 'buf.i1309'
    [5408, 5424) '_40.i.i.i1247'
    [5440, 5488) '_34.i.i.i1248'
    [5520, 5524) '_25.i.i.i1249'
    [5536, 5568) '_20.i.i.i1250'
    [5600, 5648) '_16.i.i.i1251'
    [5680, 5744) 'formatter.i.i1252'
    [5776, 5800) 'buf.i.i1253'
    [5840, 5856) '_40.i.i.i1186'
    [5872, 5920) '_34.i.i.i1187'
    [5952, 5956) '_25.i.i.i1188'
    [5968, 6000) '_20.i.i.i1189'
    [6032, 6080) '_16.i.i.i1190'
    [6112, 6176) 'formatter.i.i1191'
    [6208, 6232) 'buf.i.i1192'
    [6272, 6336) 'formatter.i1171'
    [6368, 6392) 'buf.i1172'
    [6432, 6448) '_40.i.i.i1110'
    [6464, 6512) '_34.i.i.i1111'
    [6544, 6548) '_25.i.i.i1112'
    [6560, 6592) '_20.i.i.i1113'
    [6624, 6672) '_16.i.i.i1114'
    [6704, 6768) 'formatter.i.i1115'
    [6800, 6824) 'buf.i.i1116'
    [6864, 6928) 'formatter.i1095'
    [6960, 6984) 'buf.i1096'
    [7024, 7040) '_40.i.i.i1034'
    [7056, 7104) '_34.i.i.i1035'
    [7136, 7140) '_25.i.i.i1036'
    [7152, 7184) '_20.i.i.i1037'
    [7216, 7264) '_16.i.i.i1038'
    [7296, 7360) 'formatter.i.i1039'
    [7392, 7416) 'buf.i.i1040'
    [7456, 7472) '_40.i.i.i973'
    [7488, 7536) '_34.i.i.i974'
    [7568, 7572) '_25.i.i.i975'
    [7584, 7616) '_20.i.i.i976'
    [7648, 7696) '_16.i.i.i977'
    [7728, 7792) 'formatter.i.i978'
    [7824, 7848) 'buf.i.i979'
    [7888, 7904) '_40.i.i.i912'
    [7920, 7968) '_34.i.i.i913'
    [8000, 8004) '_25.i.i.i914'
    [8016, 8048) '_20.i.i.i915'
    [8080, 8128) '_16.i.i.i916'
    [8160, 8224) 'formatter.i.i917'
    [8256, 8280) 'buf.i.i918'
    [8320, 8384) 'formatter.i897'
    [8416, 8440) 'buf.i898'
    [8480, 8544) 'formatter.i882'
    [8576, 8600) 'buf.i883'
    [8640, 8704) 'formatter.i867'
    [8736, 8760) 'buf.i868'
    [8800, 8816) '_40.i.i.i806'
    [8832, 8880) '_34.i.i.i807'
    [8912, 8916) '_25.i.i.i808'
    [8928, 8960) '_20.i.i.i809'
    [8992, 9040) '_16.i.i.i810'
    [9072, 9136) 'formatter.i.i811'
    [9168, 9192) 'buf.i.i812'
    [9232, 9248) '_40.i.i.i745'
    [9264, 9312) '_34.i.i.i746'
    [9344, 9348) '_25.i.i.i747'
    [9360, 9392) '_20.i.i.i748'
    [9424, 9472) '_16.i.i.i749'
    [9504, 9568) 'formatter.i.i750'
    [9600, 9624) 'buf.i.i751'
    [9664, 9680) '_40.i.i.i684'
    [9696, 9744) '_34.i.i.i685'
    [9776, 9780) '_25.i.i.i686'
    [9792, 9824) '_20.i.i.i687'
    [9856, 9904) '_16.i.i.i688'
    [9936, 10000) 'formatter.i.i689'
    [10032, 10056) 'buf.i.i690'
    [10096, 10112) '_40.i.i.i623'
    [10128, 10176) '_34.i.i.i624'
    [10208, 10212) '_25.i.i.i625'
    [10224, 10256) '_20.i.i.i626'
    [10288, 10336) '_16.i.i.i627'
    [10368, 10432) 'formatter.i.i628'
    [10464, 10488) 'buf.i.i629'
    [10528, 10544) '_40.i.i.i562'
    [10560, 10608) '_34.i.i.i563'
    [10640, 10644) '_25.i.i.i564'
    [10656, 10688) '_20.i.i.i565'
    [10720, 10768) '_16.i.i.i566'
    [10800, 10864) 'formatter.i.i567'
    [10896, 10920) 'buf.i.i568'
    [10960, 10976) '_40.i.i.i501'
    [10992, 11040) '_34.i.i.i502'
    [11072, 11076) '_25.i.i.i503'
    [11088, 11120) '_20.i.i.i504'
    [11152, 11200) '_16.i.i.i505'
    [11232, 11296) 'formatter.i.i506'
    [11328, 11352) 'buf.i.i507'
    [11392, 11408) '_40.i.i.i440'
    [11424, 11472) '_34.i.i.i441'
    [11504, 11508) '_25.i.i.i442'
    [11520, 11552) '_20.i.i.i443'
    [11584, 11632) '_16.i.i.i444'
    [11664, 11728) 'formatter.i.i445'
    [11760, 11784) 'buf.i.i446'
    [11824, 11840) '_40.i.i.i'
    [11856, 11904) '_34.i.i.i'
    [11936, 11940) '_25.i.i.i'
    [11952, 11984) '_20.i.i.i'
    [12016, 12064) '_16.i.i.i'
    [12096, 12160) 'formatter.i.i'
    [12192, 12216) 'buf.i.i'
    [12256, 12320) 'formatter.i415'
    [12352, 12376) 'buf.i416'
    [12416, 12480) 'formatter.i400'
    [12512, 12536) 'buf.i401'
    [12576, 12640) 'formatter.i385'
    [12672, 12696) 'buf.i386'
    [12736, 12800) 'formatter.i370'
    [12832, 12856) 'buf.i371'
    [12896, 12960) 'formatter.i355'
    [12992, 13016) 'buf.i356'
    [13056, 13120) 'formatter.i'
    [13152, 13176) 'buf.i345'
    [13216, 13240) 'buf.i'
    [13280, 13320) 'e.i.i'
    [13360, 13384) '_2.i.i'
    [13424, 13440) '_7.sroa.6.sroa.0.i' (line 651)
    [13456, 13504) 'self.i' (line 652)
    [13536, 13560) 's.i' (line 652)
    [13600, 13608) 'x169'
    [13632, 13636) 'x168'
    [13648, 13650) 'x167'
    [13664, 13712) '_397'
    [13744, 13768) '_396.sroa.0'
    [13808, 13848) 'v165'
    [13888, 13936) '_388'
    [13968, 13992) '_387.sroa.0'
    [14032, 14072) 'v162'
    [14112, 14160) '_379'
    [14192, 14216) '_378.sroa.0'
    [14256, 14280) 'v159'
    [14320, 14368) '_370'
    [14400, 14424) '_369.sroa.0'
    [14464, 14552) 'v156'
    [14592, 14640) '_361'
    [14672, 14696) '_360.sroa.0'
    [14736, 14784) 'v153'
    [14816, 14864) '_352'
    [14896, 14920) '_351.sroa.0'
    [14960, 14984) 'v150'
    [15024, 15072) '_343'
    [15104, 15128) '_342.sroa.0'
    [15168, 15216) '_339'
    [15248, 15272) '_338.sroa.0'
    [15312, 15360) '_335'
    [15392, 15416) '_334.sroa.0'
    [15456, 15504) '_331'
    [15536, 15560) '_330.sroa.0'
    [15600, 15648) '_327'
    [15680, 15704) '_326.sroa.0'
    [15744, 15792) '_323'
    [15824, 15848) '_322.sroa.0'
    [15888, 15936) '_319'
    [15968, 15992) '_318.sroa.0'
    [16032, 16080) '_315'
    [16112, 16136) '_314.sroa.0'
    [16176, 16224) '_311'
    [16256, 16280) '_310.sroa.0'
    [16320, 16368) '_307'
    [16400, 16424) '_306.sroa.0'
    [16464, 16512) '_303'
    [16544, 16568) '_302.sroa.0'
    [16608, 16656) '_299'
    [16688, 16712) '_298.sroa.0'
    [16752, 16800) '_295'
    [16832, 16856) '_294.sroa.0'
    [16896, 16944) '_291'
    [16976, 17000) '_290.sroa.0'
    [17040, 17088) '_287'
    [17120, 17144) '_286.sroa.0'
    [17184, 17200) 'v'
    [17216, 17240) '_278'
    [17280, 17304) '_277.sroa.0'
    [17344, 17360) 'x131'
    [17376, 17384) 'x129'
    [17408, 17412) 'x128'
    [17424, 17436) 'x127'
    [17456, 17468) 'x125'
    [17488, 17496) 'x123'
    [17520, 17524) 'x122'
    [17536, 17572) 'x121' <== Memory access at offset 17572 overflows this variable
    [17616, 17620) 'x119'
    [17632, 17640) 'x118'
    [17664, 17668) 'x117'
    [17680, 17682) 'x116'
    [17696, 17744) '_205'
    [17776, 17800) '_204.sroa.0'
    [17840, 17856) 'raw_out111.sroa.5' (line 782)
    [17872, 17896) '_146' (line 780)
    [17936, 17960) '_138' (line 775)
    [18000, 18024) '_132' (line 773)
    [18064, 18080) '_127' (line 768)
    [18096, 18144) '_123' (line 768)
    [18176, 18200) 'res'
    [18240, 18264) 'self110' (line 766)
    [18304, 18352) 'self106' (line 752)
    [18384, 18424) 'self105' (line 749)
    [18464, 18512) 'self104' (line 746)
    [18544, 18584) 'self103' (line 743)
    [18624, 18672) 'self102' (line 740)
    [18704, 18728) 'self101' (line 737)
    [18768, 18816) 'self100' (line 734)
    [18848, 18936) 'self99' (line 731)
    [18976, 19024) 'self98' (line 728)
    [19056, 19104) 'self97' (line 727)
    [19136, 19184) 'self96' (line 724)
    [19216, 19240) 'self95' (line 723)
    [19280, 19328) 'self94' (line 720)
    [19360, 19408) 'self93' (line 717)
    [19440, 19488) 'self92' (line 714)
    [19520, 19568) 'self91' (line 711)
    [19600, 19648) 'self90' (line 708)
    [19680, 19728) 'self89' (line 705)
    [19760, 19808) 'self88' (line 702)
    [19840, 19888) 'self87' (line 699)
    [19920, 19968) 'self86' (line 696)
    [20000, 20048) 'self85' (line 693)
    [20080, 20128) 'self84' (line 690)
    [20160, 20208) 'self83' (line 687)
    [20240, 20288) 'self82' (line 684)
    [20320, 20368) 'self81' (line 681)
    [20400, 20448) 'self80' (line 678)
    [20480, 20497) 'self79' (line 677)
    [20544, 20576) 'self78' (line 676)
    [20608, 20632) 'self77' (line 675)
    [20672, 20684) 'self76' (line 672)
    [20704, 20720) 'self74' (line 666)
    [20736, 20752) 'self73' (line 663)
    [20768, 20808) 'self70' (line 658)
    [20848, 20872) 'self64' (line 649)
    [20912, 20960) 'self63' (line 646)
    [20992, 21016) 'self61' (line 643)
    [21056, 21080) 'self60' (line 642)
    [21120, 21144) 'value' (line 642)
    [21184, 21192) 'ty' (line 641)
    [21216, 21240) 'raw_out' (line 639)
    [21280, 21304) 'out' (line 638)
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow (/usr/local/bin/testdrive+0xcd6b2d8) (BuildId: 2bba89f41fc100da) in decNumberGetBCD
Shadow bytes around the buggy address:
  0xffff95264200: f8 f8 f8 f8 f8 f8 f2 f2 f2 f2 f8 f8 f8 f2 f2 f2
  0xffff95264280: f2 f2 f8 f8 f8 f8 f8 f8 f2 f2 f2 f2 f8 f8 f8 f2
  0xffff95264300: f2 f2 f2 f2 f8 f8 f2 f2 f8 f8 f8 f2 f2 f2 f2 f2
  0xffff95264380: f8 f8 f8 f2 f2 f2 f2 f2 f8 f8 f2 f2 f8 f2 f2 f2
  0xffff95264400: f8 f2 f8 f8 f2 f2 f8 f8 f2 f2 f8 f2 f2 f2 f8 f2
=>0xffff95264480: 00 00 00 00[04]f2 f2 f2 f2 f2 f8 f2 f8 f2 f2 f2
  0xffff95264500: f8 f2 f8 f2 f8 f8 f8 f8 f8 f8 f2 f2 f2 f2 f8 f8
  0xffff95264580: f8 f2 f2 f2 f2 f2 f8 f8 f2 f2 f8 f8 f8 f2 f2 f2
  0xffff95264600: f2 f2 f8 f8 f8 f2 f2 f2 f2 f2 f8 f8 f8 f2 f2 f2
  0xffff95264680: f2 f2 f8 f8 f2 f2 f8 f8 f8 f8 f8 f8 f2 f2 f2 f2
  0xffff95264700: 00 00 00 f2 f2 f2 f2 f2 f8 f8 f8 f2 f2 f2 f2 f2
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==50099==ABORTING
```
This code is rather old https://github.com/MaterializeInc/materialize/pull/16541 @sploiselle Are you still interested in looking into this or should we take it in the QA team? Seen in https://buildkite.com/materialize/nightlies/builds/6662#018df3dc-cc00-4f7c-801b-8c53ff3cc2d3

ci-regexp: AddressSanitizer: stack-buffer-overflow.*testdrive.* in decNumberGetBCD
ci-regexp: AddressSanitizer: stack-buffer-overflow.*sqllogictest.* in decNumberGetBCD

**Comment by sploiselle** [comment]
@def- If you all have the bandwidth/interest, having QA take a look would be great. This is something I can get to eventually but have a pretty intense few weeks with feature development work slated.

**Comment by benesch** [comment]
Reopening, as we'll need to cut a new release of libdecnumber and pull it in here.

**Comment by benesch** [comment]
Just released this as decnumber-sys v0.1.6: https://crates.io/crates/decnumber-sys/0.1.6

@def- can you take it from here?
